cmd/config.go

```diff
@@ -20,3 +29,72 @@ func DefaultConfig() Config {
 		MCSDAdmin: mcsdadmin.Config{},
 	}
 }
+
+// LoadConfig loads configuration from YAML file and environment variables
+func LoadConfig() (Config, error) {
+	// Initialize koanf instance
+	k := koanf.New(".")
+
+	// Load default configuration first
+	defaultConfig := DefaultConfig()
+	if err := k.Load(structs.Provider(defaultConfig, "koanf"), nil); err != nil {
+		return Config{}, err
+	}
+
+	// Try config files in config directory only
+	configFiles := []string{
+		"config/knooppunt.yaml",
+		"config/knooppunt.yml",
+	}
+	
+	for _, cf := range configFiles {
+		if _, err := os.Stat(cf); err == nil {
+			if err := k.Load(file.Provider(cf), yaml.Parser()); err != nil {
+				return Config{}, fmt.Errorf("failed to load config file %s: %w", cf, err)
+			}
+			break
+		}
+	}
+
+	// Load environment variables with KNPT_ prefix
+	if err := k.Load(env.Provider("KNPT_", ".", func(s string) string {
+		// Convert KNPT_MCSD_LOCALDIRECTORY_FHIRBASEURL to mcsd.localDirectory.fhirBaseURL
+		// First remove the prefix and convert to lowercase
+		key := strings.TrimPrefix(s, "KNPT_")
+		parts := strings.Split(key, "_")
+		
+		// Convert to camelCase path
+		result := make([]string, len(parts))
+		for i, part := range parts {
+			if i == 0 {
+				// First part is always lowercase  
+				result[i] = strings.ToLower(part)
+			} else {
+				// Subsequent parts are camelCase
+				result[i] = strings.ToLower(part[:1]) + strings.ToLower(part[1:])
+				// Special case for common config keys
+				switch strings.ToUpper(part) {
+				case "FHIRBASEURL":
+					result[i] = "fhirBaseURL"
+				case "CONFIGFILE":
+					result[i] = "configFile"
+				case "ROOTDIRECTORIES":
+					result[i] = "rootDirectories"
+				case "LOCALDIRECTORY":
+					result[i] = "localDirectory"
```

Comment:
convert everything to lowercase instead, also in the config structs (the tags, not the field names)